skrub/_string_encoder.py

```diff
@@ -161,10 +166,11 @@ def fit_transform(self, X, y=None):
         self._is_fitted = True
         self.n_components_ = result.shape[1]
 
-        name = sbd.name(X)
-        if not name:
-            name = "tsvd"
-        self.all_outputs_ = [f"{name}_{idx}" for idx in range(self.n_components_)]
+        self._input_name = sbd.name(X)
+        if not self._input_name:
+            self._input_name = "tsvd"
```

Comment:
Simplifies:
```suggestion
        self._input_name = sbd.name(X) or "tsvd"
```

This default naming is a good idea, we could do the same in `TextEncoder`, `MinHashEncoder`, `GapEncoder`, WDYT?

Reply:
yes we could do that, but maybe in a separate pr